app/forms/providers/means/regular_income_form.rb

```diff
@@ -0,0 +1,169 @@
+module Providers
+  module Means
+    class RegularIncomeForm
+      include ActiveModel::Model
+
+      INCOME_TYPES = %w[
+        benefits
+        friends_or_family
+        maintenance_in
+        property_or_lodger
+        student_loan
+        pension
+      ].freeze
+
+      attr_reader :transaction_type_ids, :legal_aid_application
+
+      INCOME_TYPES.each do |income_type|
+        attr_accessor "#{income_type}_amount".to_sym,
+                      "#{income_type}_frequency".to_sym
+      end
+
+      validates :transaction_type_ids, presence: true, unless: :none_selected?
+      validate :all_regular_transactions_valid
+
+      def initialize(params = {})
+        @none_selected = none_selected.in?(params["transaction_type_ids"] || [])
+        @legal_aid_application = params.delete(:legal_aid_application)
+        @transaction_type_ids = params["transaction_type_ids"] || @legal_aid_application.transaction_type_ids
+
+        assign_regular_transaction_attributes
+
+        super
+      end
+
+      def save
+        return false unless valid?
+
+        ApplicationRecord.transaction do
+          destroy_transactions!
+
+          build_legal_aid_application_transaction_types
+          legal_aid_application.assign_attributes(
+            no_credit_transaction_types_selected: none_selected?,
+          )
+          legal_aid_application.save!
+
+          regular_transactions.each(&:save!)
+        end
+
+        true
+      end
+
+      def transaction_type_ids=(ids)
+        @transaction_type_ids = if none_selected?
+                                  []
+                                else
+                                  ids.compact_blank
+                                end
+      end
+
+      def transaction_type_options
+        TransactionType.not_children.credits
+      end
+
+      def none_selected
+        "none"
+      end
+
+    private
+
+      def assign_regular_transaction_attributes
+        regular_transactions.each do |transaction|
+          transaction_type = transaction.transaction_type
+          public_send("#{transaction_type.name}_amount=", transaction.amount)
+          public_send("#{transaction_type.name}_frequency=", transaction.frequency)
+        end
+      end
+
+      def none_selected?
+        @none_selected
+      end
+
+      def transaction_types
+        transaction_type_options.where(id: transaction_type_ids)
+      end
+
```

Comment:
My only thought is that perhaps these methods could be extracted into a helper class, as some of this functionality will be replicated on the outgoings page as well as on the current `identify_types_of_income/outgoing` and the cash transactions pages. We currently have the TransactionType helper - maybe this would be a suitable place? I'm not suggesting this needs be done in this PR though, just a suggestion for later on

Reply:
We might do, but we might not.

I'd rather wait for the right abstractions to present themselves clearly, rather than try and predict what the abstractions might be to "DRY" things up —and potentially making things more difficult for ourselves.

I propose we leave it for now.

Does that sound ok to you?